src/azure-cli-core/azure/cli/core/command_recommender.py

```diff
@@ -211,66 +230,65 @@ def _disable_aladdin_service(self):
-                '--output': None,
-                '--query': None,
-                '--debug': None,
-                '--verbose': None
-            }
-
-            if parameter_table:
-                for argument in parameter_table.values():
-                    options = argument.type.settings['options_list']
-                    options = [option for option in options if not isinstance(option, Deprecated)]
-                    # skip the positional arguments
-                    if not options:
-                        continue
-                    try:
-                        sorted_options = sorted(options, key=len, reverse=True)
-                        standard_form = sorted_options[0]
-
-                        for option in sorted_options[1:]:
-                            rules[option] = standard_form
-                        rules[standard_form] = None
-                    except TypeError:
-                        logger.debug('Unexpected argument options `%s` of type `%s`.', options, type(options).__name__)
-
-            for parameter in parameters:
-                if parameter in rules:
-                    normalized_form = rules.get(parameter, None) or parameter
-                    normalized_parameters.append(normalized_form)
-                else:
-                    logger.debug('"%s" is an invalid parameter for command "%s".', parameter, self.command)
+        param_mappings = {
+            '-h': '--help',
+            '-o': '--output',
+            '--only-show-errors': None,
+            '--help': None,
+            '--output': None,
+            '--query': None,
+            '--debug': None,
+            '--verbose': None
+        }
+
+        if parameter_table:
+            for argument in parameter_table.values():
+                options = argument.type.settings['options_list']
+                options = [option for option in options if not isinstance(option, Deprecated)]
+                # skip the positional arguments
+                if not options:
+                    continue
+                try:
+                    sorted_options = sorted(options, key=len, reverse=True)
+                    standard_form = sorted_options[0]
+
+                    for option in sorted_options[1:]:
+                        param_mappings[option] = standard_form
+                    param_mappings[standard_form] = standard_form
+                except TypeError:
+                    logger.debug('Unexpected argument options `%s` of type `%s`.', options, type(options).__name__)
+
+        return param_mappings
+
+    def _normalize_parameters(self, raw_parameters):
+        """Normalize a parameter list.
+        Get the standard parameter names of the raw parameters, which includes:
+            1. Use long options to replace short options
+            2. Remove the unrecognized parameter names
+        An example: ['-g', 'RG', '-n', 'NAME'] ==> {'--resource-group': 'RG', '--name': 'NAME'}
+        """
```

Comment:
Good job! This can be used to solve this [telemetry issue](https://github.com/Azure/azure-cli/issues/10170), just assigned it to you.